ct/cmd/install.go

```diff
@@ -31,13 +31,17 @@ func newInstallCmd() *cobra.Command {
 		Use:   "install",
 		Short: "Install and test a chart",
 		Long: heredoc.Doc(`
-			Run 'helm install' and ' helm test' on
+			Run 'helm install', 'helm test', and optionally 'helm upgrade' on
 
 			* changed charts (default)
 			* specific charts (--charts)
 			* all charts (--all)
 
-			in given chart directories.
+			in given chart directories. If upgrade (--upgrade) is true, then this
+			command will validate that 'helm test' passes for the following upgrade paths:
+
+			* previous chart revision => current chart version (if non-breaking SemVer change)
```

Comment:
Should this be "previous chart revisions" plural?

Reply:
Never mind, I see it is not, but see my note above – I believe it should be.

Reply:
Re testing _all_ previous semver-compatible revisions:

Yes this would be ideal, though I do worry about the impact this could have on test duration. Perhaps testing the very first compatible version and the very last would be a good enough signal? Can you think of any cases where this would break down?

Reply:
An example is if someone broke then fixed one of the label immutability issues (or some other breaking change) without bumping the MAJOR version. In this case, one or more MINOR versions may fail the next upgrade even if the first and last one pass.

If we test upgrades between individual releases, issues like this should be caught moving forward… but it won't help to fix any problems that exist before we start testing upgrades between each release.

Practically speaking this may be too much for our testing large chart repos (like https://github.com/helm/charts). Good idea to meet, let's discuss Tomorrow and decide.

Reply:
Update: we discussed this and agreed the use case I mentioned should only need to be run once per repo, if thereafter the most recent release is tested for upgrades. Therefore, if we care to do this for a historic list (to catch up on an existing repo) it could either be scripted manually, or else added as a new single-use function in a follow-up PR.